internal/database/types_hcpcluster.go

```diff
@@ -0,0 +1,96 @@
+package database
+
+import (
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+	azcorearm "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+)
+
+// HCPClusterDocument represents a customer desired HCP Cluster.
+// To transition from our current state using cluster-service as half the source of truth to a state where
+// cosmos contains all the desired state and all the observed state, we are basing the schema on ResourceDocument.
+type HCPClusterDocument struct {
+	ResourceDocument `json:",inline"`
+
+	// CustomerDesiredState represents information from the user.  Some minimal validation may happen prior to persisting
+	// this information, but it is entirely possible that this state does not reflect an achievable reality.
+	// This information is necessary to determine the HostedCluster and supporting resources that are desired.
+	CustomerDesiredState CustomerDesiredState `json:"customerDesiredState"`
+
+	ServiceProviderState ServiceProviderState `json:"serviceProviderState"`
+}
+
+func NewHCPClusterDocument(resourceID *azcorearm.ResourceID) *HCPClusterDocument {
+	return &HCPClusterDocument{
+		ResourceDocument: ResourceDocument{
+			ResourceID: resourceID,
+		},
+	}
+}
+
+// GetValidTypes returns the valid resource types for a ResourceDocument.
+func (doc HCPClusterDocument) GetValidTypes() []string {
+	return []string{
+		api.ClusterResourceType.String(),
+	}
+}
```

Comment:
I think you'll want to remove `api.ClusterResourceType` from `ResourceDocument.GetValidTypes` since `HCPClusterDocument` is taking over.  Might not matter for execution but just for clarity.